### PR TITLE
Add service priority flag to kgs template cluster

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/giantswarm/issues/22334

As the creator of a pull request, please consider:

- [ ] Describe the goal you are trying to accomplish here, above the line.
- [ ] Add a user-friendly description of your change to `CHANGELOG.md`.
- [ ] Provide a link to the issue you are solving or working towards, if available.
- [ ] Request SIG UX for review. They care about almost all user-facing changes.
- [ ] Update the public [kubectl-gs documentation](https://docs.giantswarm.io/ui-api/kubectl-gs/) to reflect the changes here.
- [ ] add the `breaking-change` label to the PR if the change you are making changes the existing behaviour. Examples: removal of a flag, removal of a command, change of a default value. (Such changes should be released with a **major version** bump.)

Feel free to remove this checklist when done.

### CODE REVIEW

**Review by kuosandys — APPROVED** [review]
Tested templating clusters with the `--service-priority` flag, and it works great for both Azure and AWS 🙂 

Just checking my understanding, we don't want to allow setting this label using the `--label` flag until the Kyverno policy is in place, right?

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -252,6 +254,7 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.PodsCIDR, flagPodsCIDR, "", "CIDR used for the pods.")
 	cmd.Flags().StringVar(&f.Release, flagRelease, "", "Workload cluster release.")
 	cmd.Flags().StringSliceVar(&f.Label, flagLabel, nil, "Workload cluster label.")
+	cmd.Flags().StringVar(&f.ServicePriority, flagServicePriority, "highest", fmt.Sprintf("Service priority of the cluster. Must be one of %v", getServicePriorities()))
```
Comment (marians) [inline cmd/template/cluster/flag.go:257]: Can we use `label.ServicePriorityHighest` here instead of the string literal?

Comment (anvddriesch) [reply cmd/template/cluster/flag.go:257]: Oh yeah, sure :)

**Comment by anvddriesch** [comment]
> Just checking my understanding, we don't want to allow setting this label using the --label flag until the Kyverno policy is in place, right?

I think it is possible to do this, right? (Unless it is blocked by another admission controller maybe)
However, we did not want to encourage it before having the policy in place.

**Comment by kuosandys** [comment]
> > Just checking my understanding, we don't want to allow setting this label using the --label flag until the Kyverno policy is in place, right?
> 
> I think it is possible to do this, right? (Unless it is blocked by another admission controller maybe) However, we did not want to encourage it before having the policy in place.

Thank you for explaining! I noticed we can't set labels with the substring `giantswarm.io` at the moment with the `--label` flag, but I wasn't sure if that was the intended effect. If I run the command with `--label="giantswarm.io/service-priority=highest` I see:
```
Error: Invalid flag: --label must contain valid label definitions (invalid label key error: "giantswarm.io/service-priority=highest": containing forbidden substring 'giantswarm.io')
```

**Comment by anvddriesch** [comment]
Ah yeah I guess that is the existing validation.

**Pull request merged**